osutil/disks/disks.go

```diff
@@ -0,0 +1,288 @@
+	// matching the specified filesystem label on the disk. Note that for
+	// non-ascii labels like "Some label", the label will be encoded using
+	// \x<hex> for potentially non-safe characters like in "Some\x20Label".
+	FindMatchingPartitionUUID(string) (string, error)
+
+	// MountPointIsFromDisk returns whether the specified mountpoint corresponds
+	// to a partition on the disk. Note that this only considers partitions
+	// and mountpoints found when the disk was identified with
+	// DiskFromMountPoint.
+	MountPointIsFromDisk(string, *Options) (bool, error)
+
+	// Dev returns the string "major:minor" number for the disk device.
+	Dev() string
+}
+
+func parseDeviceMajorMinor(s string) (int, int, error) {
+	errMsg := fmt.Errorf("invalid device number format: (expected <int>:<int>)")
+	devNums := strings.SplitN(s, ":", 2)
+	if len(devNums) != 2 {
+		return 0, 0, errMsg
+	}
+	maj, err := strconv.Atoi(devNums[0])
+	if err != nil {
+		return 0, 0, errMsg
+	}
+	min, err := strconv.Atoi(devNums[1])
+	if err != nil {
+		return 0, 0, errMsg
+	}
+	return maj, min, nil
+}
+
+var udevadmProperties = func(device string) ([]byte, error) {
+	// TODO: maybe combine with gadget interfaces hotplug code where the udev
+	// db is parsed?
+	cmd := exec.Command("udevadm", "info", "--query", "property", "--name", device)
+	return cmd.CombinedOutput()
+}
+
+func udevProperties(device string) (map[string]string, error) {
+	out, err := udevadmProperties(device)
+	if err != nil {
+		return nil, osutil.OutputErr(out, err)
+	}
+	r := bytes.NewBuffer(out)
+
+	return parseUdevProperties(r)
+}
+
+func parseUdevProperties(r io.Reader) (map[string]string, error) {
+	m := make(map[string]string)
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		strs := strings.SplitN(scanner.Text(), "=", 2)
+		if len(strs) != 2 {
+			// bad udev output?
+			continue
+		}
+		m[strs[0]] = strs[1]
+	}
+
+	return m, scanner.Err()
+}
+
+// DiskFromMountPoint finds a matching Disk for the specified mount point.
+func DiskFromMountPoint(mountpoint string, opts *Options) (Disk, error) {
+	// call the unexported version that may be mocked by tests
+	return diskFromMountPoint(mountpoint, opts)
+}
+
+type disk struct {
+	major int
+	minor int
+	// fsLabelToPartUUID is a map of filesystem label -> partition uuid for now
+	// eventually this may be expanded to be more generally useful
+	fsLabelToPartUUID map[string]string
+}
+
+// diskFromMountPointImpl returns a Disk for the underlying mount source of the
+// specified mount point. For mount points which have sources that are not
+// partitions, and thus are a part of a disk, the returned Disk refers to the
+// volume/disk of the mount point itself.
+func diskFromMountPointImpl(mountpoint string, opts *Options) (*disk, error) {
+	// first get the mount entry for the mountpoint
+	mounts, err := osutil.LoadMountInfo()
+	if err != nil {
+		return nil, err
+	}
+	found := false
+	d := &disk{}
```

Comment:
nitpick^99: we could use `var d *disk` here and and instead of found just check for `d == nil` below. But no need to change this, just an idle observation while reading the code.

Reply:
sure this is easy enough to do instead